homeassistant/components/toon/sensor.py

```diff
@@ -1,217 +1,137 @@
         return self._unit_of_measurement
 
-    def update(self):
+    async def async_update(self) -> None:
         """Get the latest data from the sensor."""
-        self.thermos.update()
+        section = getattr(self.toon, self.section)
+        value = None
 
+        if self.section == 'power' and self.measurement == 'daily_value':
+            value = round((float(section.daily_usage)
+                           + float(section.daily_usage_low)) / 1000.0, 2)
 
-class FibaroSensor(Entity):
-    """Representation of a Fibaro sensor."""
+        if value is None:
+            value = getattr(section, self.measurement)
 
-    def __init__(self, hass, name, plug_name, icon, unit_of_measurement):
-        """Initialize the Fibaro sensor."""
-        self._name = name
-        self._plug_name = plug_name
-        self._state = None
-        self._icon = 'mdi:{}'.format(icon)
-        self._unit_of_measurement = unit_of_measurement
-        self.toon = hass.data[toon_main.TOON_HANDLE]
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+        if self.section == 'power' and \
+                self.measurement in ['meter_reading', 'meter_reading_low',
+                                     'average_daily']:
+            value = round(float(value)/1000.0, 2)
 
-    @property
-    def icon(self):
-        """Return the mdi icon of the sensor."""
-        return self._icon
+        if self.section == 'solar' and \
+                self.measurement in ['meter_reading_produced',
+                                     'meter_reading_low_produced']:
+            value = float(value)/1000.0
 
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        value = '_'.join(self.name.lower().split('_')[1:])
-        return self.toon.get_data(value, self._plug_name)
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit this state is expressed in."""
-        return self._unit_of_measurement
-
-    def update(self):
-        """Get the latest data from the sensor."""
-        self.toon.update()
-
-
-class SolarSensor(Entity):
-    """Representation of a Solar sensor."""
-
-    def __init__(self, hass, name, unit_of_measurement):
-        """Initialize the Solar sensor."""
-        self._name = name
-        self._state = None
-        self._icon = 'mdi:weather-sunny'
-        self._unit_of_measurement = unit_of_measurement
-        self.toon = hass.data[toon_main.TOON_HANDLE]
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def icon(self):
-        """Return the mdi icon of the sensor."""
-        return self._icon
+        if self.section == 'gas' and \
+                self.measurement in ['average_daily', 'daily_usage',
+                                     'meter_reading']:
+            value = round(float(value)/1000.0, 2)
```

Comment:
The gas cost can not be handled the same way as the other sensors. it is already a rounded float. 
Could this be an possible solution?
```python
        if self.section == 'gas' and \
                self.measurement in ['daily_cost']:
            value = float(value)
```
I used `float(self.toon.gas.daily_cost)` in the old component. That works.
I have no way of testing this in the new code because I do not know how to enable the UI integration for custom components.

Reply:
I think you might have misread/misinterpreted the code.
The costs value is not adjusted, cast or recalculated in any way.